app/oxalis/security/CompactRandomIDGenerator.scala

```diff
@@ -9,23 +9,25 @@ import scala.concurrent.{ExecutionContext, Future}
 
 object CompactRandomIDGenerator {
   lazy val random = new SecureRandom()
+
+  def generateBlocking(sizeInBytes: Int = 16): String = {
+    val randomValue = new Array[Byte](sizeInBytes)
+    CompactRandomIDGenerator.random.nextBytes(randomValue)
+    encode(randomValue)
+  }
+
+  def encode(bytes: Array[Byte]) = Base64.getUrlEncoder.withoutPadding.encodeToString(bytes)
 }
 
 class CompactRandomIDGenerator(sizeInBytes: Int = 16)(implicit ec: ExecutionContext) extends IDGenerator {
 
   override def generate: Future[String] = {
     val randomValue = new Array[Byte](sizeInBytes)
     Future(CompactRandomIDGenerator.random.nextBytes(randomValue)).map { _ =>
-      encode(randomValue)
+      CompactRandomIDGenerator.encode(randomValue)
```

Comment:
Moved the non-blocking `generate` into a class, because it needs an ExecutionContext in scope.
The blocking version (which does not need an ec) is still in the object, so that it can be called from `object DataStoreHandler`